response.go

```diff
@@ -961,6 +992,43 @@ func (r *Response) getJSONP(
 	return value
 }
 
+// Reader returns the body reader from the response.
+//
+// Reader replaces the other methods for reading response body
+// and disables rewinding when reading.
+//
+// Example:
+//
+//	resp := NewResponse(t, response)
+//	reader := resp.Reader()
+func (r *Response) Reader() io.ReadCloser {
+	opChain := r.chain.enter("Reader()")
+	defer opChain.leave()
+
+	if opChain.failed() {
+		return newErrorReader(errors.New("cannot call Reader() because chain has failed"))
+	}
+
+	if r.contentState != contentPending {
+		err := fmt.Errorf("cannot call Reader() because %s was already called", r.contentMethod)
```

Comment:
Let's report this error (`cannot call Reader() because %s was already called`) to opChain.fail, but from we can return the same error as above (`cannot read from failed Response`).

In other words, when something goes wrong, Response reports failure to chain, marks itself failed, and returns reader that always reports error `cannot read from failed Response`.